### ISSUE TITLE
ERR_CHILD_PROCESS_STDIO_MAXBUFFER

### ISSUE THREAD

**Issue by Benjy1979** [desktop/desktop#15486]
### The problem

For certain projects I have (ember.js) the "Hang On..." keeps getting stuck. When pushing and getting the latest status. The twirling indicator churns eternally.

The log file shows lots of:
`2022-10-19T11:48:47.599Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: roomviewer-3d-data-v2 - Benjy1979 (has token)
2022-10-19T11:48:48.450Z - error: [ui] Could not load new contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
2022-10-19T11:48:48.453Z - error: [ui] Could not load old contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)`

### Release version

Version 3.1.2 (x64)

### Operating system

Windows 11

### Steps to reproduce the behavior

I'm not sure what causes it. But I have ember projects.

### Log files

[2022-10-19.desktop.production.log](https://github.com/desktop/desktop/files/9820985/2022-10-19.desktop.production.log)

**Comment by steveward** [comment]
Thanks for the report @Benjy1979. There was a range error issue reported in https://github.com/desktop/desktop/issues/15355. Is this error only happening during push operations? Is it causing GitHub Desktop to crash, or just lock up?

I know you mentioned seeing this specifically in Ember projects -- would you be able to test things out in another test repository to see if you are able to reproduce the issue? It'd also be helpful to know if any of the repositories you are encountering this in are open source -- I'd love to try and reproduce this issue directly.

**Comment by Benjy1979** [comment]
![image](https://user-images.githubusercontent.com/5563332/196764957-5c4e0be7-96d1-4a81-b17d-e88b979dae12.png)

It just causes this to run forever. Even getting the status.
Do you know what the logging errors imply by any chance? I don't use squash and merge.

I strongly doubt a new project exhibits this issue I have other similar repos, same product with slight differences which don't exhibit it.

**Comment by sergiou87** [comment]
@Benjy1979 could you run `git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin` or `git -c credential.helper= -c rebase.backend=merge pull --ff --recurse-submodules --progress origin` in that repository?

I wonder if those commands have a large output that could cause the problem. I think by default those git commands will have a 10MB buffer for allocated for stdout.

**Comment by Benjy1979** [comment]
The first command I run, the command had no output in the console it just went back to the prompt - it didn't get stuck. The second one, said:"already updated". And went back to the prompt. No halt.
It hasn't made any difference to GitHub Desktop's operation either. No output generally came out of running those commands?

**Comment by Benjy1979** [comment]
Sorry , I just checked the log here it is attached.It does look like the buffer's fault.
This might be another bug or related, but in the roomviewer-3d-data-v2 project, it won't load it sort of goes to a state where it can't "locate" the folder at all. Even though it's there. It just gives options to "retry", "clone again" etc That is a big repo with lots of files though.

`2023-02-06T09:36:31.429Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2023-02-06T09:36:32.124Z - info: [ui] [AppStore] loading 26 repositories from store
2023-02-06T09:36:32.124Z - info: [ui] [AppStore] found account: Benjy1979 (Ben Glancy)
2023-02-06T09:36:32.399Z - info: [ui] launching: 3.1.5 (Windows 10.0.22621)
2023-02-06T09:36:32.402Z - info: [ui] execPath: 'C:\Users\ben\AppData\Local\GitHubDesktop\app-3.1.5\GitHubDesktop.exe'
2023-02-06T09:36:32.593Z - info: [ui] Subscribed 'Benjy1979' to Alive channel
2023-02-06T09:36:32.743Z - info: [ui] Stats reported.
2023-02-06T09:36:50.004Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.122s)
2023-02-06T09:36:56.590Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.099s)
2023-02-06T09:37:20.511Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.089s)
2023-02-06T09:37:26.031Z - info: [ui] Executing getStashEntries: git log -g -z --pretty=%gD%x1F%H%x1F%gs refs/stash (took 5.337s)
2023-02-06T09:37:26.038Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.048s)
2023-02-06T09:37:26.038Z - info: [ui] Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 -- (took 5.044s)
2023-02-06T09:37:26.151Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.137s)
2023-02-06T09:37:41.342Z - info: [ui] No submodules found. Skipping "git submodule status"
2023-02-06T09:37:49.759Z - info: [ui] Executing getCommits: git log origin/GLB-Work..GLB-Work --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.064s)
2023-02-06T09:37:49.760Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 8d1293970a8f3b668a26c0ea7cae528d2210abb5^:app/adapters/lead.js (took 5.049s)
2023-02-06T09:37:49.842Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 8d1293970a8f3b668a26c0ea7cae528d2210abb5:app/adapters/lead.js (took 5.125s)
2023-02-06T09:37:53.300Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: ember-roomviewer-bacomponents - Benjy1979 (has token)
2023-02-06T09:38:25.816Z - info: [ui] Executing getCommits: git log origin/GLB-Work..GLB-Work --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.095s)
2023-02-06T09:39:32.446Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.095s)
2023-02-06T09:39:47.924Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 8d1293970a8f3b668a26c0ea7cae528d2210abb5^:app/services/attachments.js (took 5.092s)
2023-02-06T09:39:54.152Z - info: [ui] Executing getChangesFiles: git log cbc4dca15b42187ecc7735e6d681165312450fcc -C -M -m -1 --no-show-signature --first-parent --raw --format=format: --numstat -z -- (took 10.259s)
2023-02-06T09:39:54.218Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 8d1293970a8f3b668a26c0ea7cae528d2210abb5^:app/routes/user/planner/submit-form.js (took 5.118s)
2023-02-06T09:39:55.419Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 8d1293970a8f3b668a26c0ea7cae528d2210abb5:app/controllers/user/planner/submit-form.js (took 5.099s)
2023-02-06T09:40:36.858Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.118s)
2023-02-06T09:42:45.482Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 1.039s)
2023-02-06T09:42:51.804Z - info: [ui] Executing getRemotes: git remote -v (took 5.112s)
2023-02-06T10:43:13.838Z - info: [ui] Executing getStashEntries: git log -g -z --pretty=%gD%x1F%H%x1F%gs refs/stash (took 5.098s)
2023-02-06T10:48:27.232Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.025s)
2023-02-06T10:48:33.907Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.098s)
2023-02-06T10:48:39.092Z - info: [ui] Executing getCommits: git log origin/master..master --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.019s)
2023-02-06T10:48:39.092Z - info: [ui] Executing getRemotes: git remote -v (took 5.011s)
2023-02-06T10:48:39.174Z - info: [ui] Executing getStashEntries: git log -g -z --pretty=%gD%x1F%H%x1F%gs refs/stash (took 5.090s)
2023-02-06T10:48:44.183Z - info: [ui] Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT (took 5.097s)
2023-02-06T10:50:24.902Z - info: [ui] Executing getStashEntries: git log -g -z --pretty=%gD%x1F%H%x1F%gs refs/stash (took 5.100s)
2023-02-06T11:14:25.943Z - info: [ui] [Timing] Action 'create commit' for 'Benjy1979/ember-roomviewer-trend' took 0.635s
2023-02-06T11:14:26.572Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: ember-roomviewer-trend - Benjy1979 (has token)
2023-02-06T11:14:27.937Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.359s)
2023-02-06T11:14:35.263Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 5.093s)
2023-02-06T11:14:40.532Z - info: [ui] Executing getCommits: git log origin/master..master --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.099s)
2023-02-06T12:27:36.826Z - info: [ui] Executing getCommits: git log origin/master..master --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color -- (took 5.085s)
2023-02-06T12:50:23.551Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.163s)
2023-02-06T13:11:17.335Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.668s)
2023-02-06T13:11:18.595Z - info: [ui] Executing getRemotes: git remote -v (took 5.101s)
2023-02-06T13:11:29.695Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:11:29.705Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:11:35.987Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:11:39.164Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:11:40.944Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:15:35.374Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2023-02-06T13:15:35.665Z - info: [ui] [AppStore] loading 26 repositories from store
2023-02-06T13:15:35.665Z - info: [ui] [AppStore] found account: Benjy1979 (Ben Glancy)
2023-02-06T13:15:35.765Z - info: [ui] launching: 3.1.5 (Windows 10.0.22621)
2023-02-06T13:15:35.766Z - info: [ui] execPath: 'C:\Users\ben\AppData\Local\GitHubDesktop\app-3.1.5\GitHubDesktop.exe'
2023-02-06T13:15:36.044Z - info: [ui] Subscribed 'Benjy1979' to Alive channel
2023-02-06T13:15:43.677Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:15:43.686Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:15:45.454Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:40:54.154Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:40:54.214Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: ember-roomviewer-bacomponents - Benjy1979 (has token)
2023-02-06T13:40:56.330Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:11.016Z - info: [ui] [AppStore] adding repository at C:\GitHub\roomviewer-3d-data-v2 to store
2023-02-06T13:41:15.972Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:17.520Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:33.643Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:46.417Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:53.011Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:41:55.664Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:43:14.502Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/Benjy1979/roomviewer-3d-data-v2.git - Benjy1979 (has token)
2023-02-06T13:43:14.694Z - error: [ui] `git -c credential.helper= -c init.defaultBranch=main clone --recursive --progress -- https://github.com/Benjy1979/roomviewer-3d-data-v2.git C:\GitHub\roomviewer-3d-data-v2` exited with an unexpected code: 128.
stderr:
fatal: destination path 'C:\GitHub\roomviewer-3d-data-v2' already exists and is not an empty directory.

2023-02-06T13:43:19.262Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:41.397Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2023-02-06T13:48:41.727Z - info: [ui] [AppStore] loading 26 repositories from store
2023-02-06T13:48:41.728Z - info: [ui] [AppStore] found account: Benjy1979 (Ben Glancy)
2023-02-06T13:48:41.837Z - info: [ui] launching: 3.1.5 (Windows 10.0.22621)
2023-02-06T13:48:41.844Z - info: [ui] execPath: 'C:\Users\ben\AppData\Local\GitHubDesktop\app-3.1.5\GitHubDesktop.exe'
2023-02-06T13:48:42.094Z - info: [ui] Subscribed 'Benjy1979' to Alive channel
2023-02-06T13:48:48.035Z - info: [ui] [BranchPruner] Last prune took place 38 minutes ago - skipping
2023-02-06T13:48:50.598Z - error: [ui] Could not load new contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
2023-02-06T13:48:52.156Z - error: [ui] Could not load old contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
2023-02-06T13:48:52.164Z - error: [ui] Could not load new contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
2023-02-06T13:48:54.279Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:54.291Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:55.586Z - info: [ui] Executing getPartialBlobContentsCatchPathNotInRef: git show 5bec6269137c1ec58e94783e65ad5531ae032f32^:3d-app/ArtiCAD_Units.js (took 5.094s)
2023-02-06T13:48:55.587Z - error: [ui] Could not load old contents for syntax highlighting
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at Socket.onChildStdout (node:child_process:470:14)
    at Socket.emit (node:events:538:35)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
2023-02-06T13:48:58.545Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:58.556Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:58.566Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:48:58.574Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:49:03.840Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:49:20.274Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:49:20.282Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:49:23.921Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:52:15.148Z - info: [main] Using toast activator CLSID {27D44D0C-A542-5B90-BCDB-AC3126048BA2}
2023-02-06T13:52:15.454Z - info: [ui] [AppStore] loading 26 repositories from store
2023-02-06T13:52:15.455Z - info: [ui] [AppStore] found account: Benjy1979 (Ben Glancy)
2023-02-06T13:52:15.550Z - info: [ui] launching: 3.1.5 (Windows 10.0.22621)
2023-02-06T13:52:15.553Z - info: [ui] execPath: 'C:\Users\ben\AppData\Local\GitHubDesktop\app-3.1.5\GitHubDesktop.exe'
2023-02-06T13:52:15.811Z - info: [ui] Subscribed 'Benjy1979' to Alive channel
2023-02-06T13:52:22.565Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: ember-roomviewer-bacomponents - Benjy1979 (has token)
2023-02-06T13:52:23.523Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:52:23.535Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes
2023-02-06T13:52:59.581Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 25 repositories took 11.1s of which 27.4s paused, total 38.5s
2023-02-06T13:54:33.683Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 5.016s)
2023-02-06T13:54:33.702Z - info: [ui] Executing getRemotes: git remote -v (took 5.038s)
2023-02-06T13:54:33.704Z - info: [ui] [BranchPruner] Last prune took place 2 minutes ago - skipping
`

**Comment by sergiou87** [comment]
> That is a big repo with lots of files though

This is probably the reason, then :( You can also see here Desktop is having trouble to process the whole `git status` output:

> 2023-02-06T13:48:58.545Z - error: [ui] 'git status' emitted 24004090 bytes, which is beyond the supported threshold of 20000000 bytes

Still, it shouldn't get stuck, it should show an error or something like that instead 😕 Do you have any public repository we could use to reproduce this and try to figure out what the problem is?

**Comment by Benjy1979** [comment]
That might be a pain for little rewards because I have the same REPO in another two locations and it doesn't exhibit the crash on the other machines. Is there a set of temporary files I can clear that GitHub desktop writes to? I'm wondering if a fie GitHub Desktop uses is corrupted perhaps.

**Comment by sergiou87** [comment]
Oops! Sorry it took a while to get back to you. Crazy weeks 😅 

Yes, you can delete/move the `C:\Users<YourUserName>\AppData\Roaming\GitHub Desktop` folder. This folder has all your GitHub Desktop settings and the list of repositories.

If the same repo is in other two locations and works fine there, I'd even try something like removing/re-adding the repo to desktop, moving it to another path… anything you can think of that could make a difference. If you let me know what worked, maybe that gives me a clue of what could be wrong 🤔

**Comment by Benjy1979** [comment]
There was enough information in this ticket for it to be resolved. Bottom line is GitHub Desktop cannot handle a lot of output from Git. Instead of clipping it and getting on with, it crashes when trying to comprehend the repo.

**Comment by matttdawson** [comment]
I have the same issue now.  Could it be related to a husky task?  As for me, I get the error, then it takes a few minutes to commit, I suspect because the husky validation is running in the background.  Once it completes it fetches the repo again and all is fine.

**Comment by steveward** [comment]
Reported again in https://github.com/desktop/desktop/issues/19020.

**Comment by niik** [comment]
:wave: Great news, we believe we've addressed the source of these errors.

The maxbuffer issue when loading content for syntax highlighting was fixed in https://github.com/desktop/desktop/pull/19471
The 20Mb limit on git status output was removed in https://github.com/desktop/desktop/pull/19583

With that I think we can consider this issue closed. There are likely other commands which could generate large output but we'll deal with them in separate issues as we encounter them.

We expect these changes to make it into the next production release which should ship within a month or so
